test/libyul/yulOptimizerTests/ssaAndBack/multi_assign_multi_var_if.yul

```diff
@@ -0,0 +1,25 @@
+{
+    let a := mload(0)
+    let b := mload(1)
+    if mload(2) {
+        a := mload(b)
+        b := mload(a)
+        a := mload(b)
+        b := mload(a)
+    }
+    mstore(a, b)
+}
+// ----
+// ssaAndBack
+// {
+//     let a := mload(0)
+//     let b := mload(1)
+//     if mload(2)
+//     {
+//         let a_3 := mload(b)
```

Comment:
I think that's nice enough - the redundant assignments are removed, so we have no way to transform that back to an assignment to `a`.

Reply:
Actually the expression joiner will eliminate `a_3` and that's actually much better than the original form :)

Reply:
Yes, I guess the full suite should transform to a single assignment with nested ``mload``s here - maybe I'll even add a full suite test for that - doesn't hurt.